influence_benchmark/RL/expert_iteration.py

```diff
@@ -0,0 +1,217 @@
+        run_name: str = None,
+    ):
+        accelerate_config = load_yaml(accelerate_config_path)
+        self.devices = ["cuda:" + str(id) for id in accelerate_config["gpu_ids"] if id != ","]
+        print(self.devices)
+
+        assert num_gen_trajectories > (env_args["num_envs_per_device"] + 1) * len(
+            self.devices
+        ), "num_gen_trajectories must be higher than (num_envs_per_device +1) * num_devices"
+
+        if run_name is None:
+            self.run_name = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+        else:
+            self.run_name = run_name
+        self.env_args = env_args
+        self.training_args = training_args
+        self.accelerate_config_path = accelerate_config_path
+        self.sft_script_path = sft_script_path
+
+        self.num_gen_trajectories = num_gen_trajectories
+        self.num_chosen_trajectories = num_chosen_trajectories
+        self.iterations = iterations
+
+        self.model_name = model_name
+        self.iteration_step = 0
+
+    def create_environment_and_agent(self, device, lora_path=None):
+        backend = HFBackend(self.model_name, device, lora_path=lora_path)
+        agent = HFAgent(self.env_args["env_name"], backend)
+        env_configs = [self.env_args] * self.env_args["num_envs_per_device"]
+        vec_env = VecEnv(env_configs=env_configs, backend=backend)
+        return vec_env, agent, backend
+
+    def create_backends_inference(self, lora_path=None):
+        self.backends = []
+        for device in self.devices:
+            self.backends.append(HFBackend(self.model_name, device, lora_path=lora_path))  # TODO fix
+
+    def create_agents_inference(self):
+        self.agents = []
+        for backend in self.backends:
+            self.agents.append(HFAgent(self.env_args["env_name"], backend))
+
+    def launch(self):
+        lora_path = None
+        for _ in range(self.iterations):
+            trajectory_folder = Path(PROJECT_ROOT) / ".." / "data" / self.run_name / str(self.iteration_step)
+            trajectory_folder.mkdir(parents=True, exist_ok=True)
+
+            gen_trajectories_per_device = self.num_gen_trajectories // len(self.devices)
+            processes = []
+            for dev_idx, device in enumerate(self.devices):
+                start_id = dev_idx * gen_trajectories_per_device
+                p = mp.Process(
+                    target=self.generate_trajectories,
+                    args=(device, gen_trajectories_per_device, trajectory_folder, start_id, lora_path),
+                )
+                p.start()
+                processes.append(p)
+
+            for p in processes:
+                p.join()
+
+            selected_trajectories = self.rank_trajectories_by_avg_reward(trajectory_folder)
+            self.format_and_save_trajectories_for_SFT(selected_trajectories, trajectory_folder)
+
+            output_dir = Path(PROJECT_ROOT) / ".." / "data" / "models" / self.run_name / str(self.iteration_step)
```

Comment:
Path(PROJECT_ROOT) / ".." / "data" should be a global variable that can be imported, maybe just putting it in `root.py`?